### PR TITLE
add fragment support

### PR DESCRIPTION
**Pull request by vankop**

### CODE REVIEW

**Comment by vankop** [comment]
I followed browser Url..
`new URL("http://google.com#foo").hash = '#foo'`

**Comment by sokra** [comment]
> I followed browser Url..
> `new URL("http://google.com#foo").hash = '#foo'`

hmm, yeah. I think thats for history reasons. `URL.search` returns the query. That naming is weird.

The spec for that also calls it fragment:

> The `hash` getter steps are:
> If this’s URL’s fragment is either null or the empty string, then return the empty string.
> Return U+0023 (#), followed by this’s URL’s fragment.

File: lib/Resolver.js [hunk 1]
```diff
@@ -363,23 +380,58 @@ class Resolver {
...
+		if (i === identifier.length) {
+			if (idxQuery === 0) {
+				part.query = identifier;
+			} else if (idxQuery > 0) {
+				part.request = identifier.slice(0, idxQuery);
+				part.query = identifier.slice(idxQuery);
+			} else {
+				part.request = identifier;
+			}
+		}
```
[the remark this answers is not in the document]
Comment (vankop) [reply lib/Resolver.js:433]: I thought in this part of `enhanced-resolve` performance matter 😞

[the remark this answers is not in the document]
Comment (sokra) [reply lib/Resolver.js]: Regexp is also pretty fast if correctly written

**Comment by sokra** [comment]
Thanks

**Pull request merged**